x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsProcessManager.java

```diff
@@ -139,6 +142,8 @@ private void writeDataRows(DataFrameDataExtractor dataExtractor, AnalyticsProces
                         process.writeRecord(record);
                     }
                 }
+                rowsProcessed += rows.get().size();
+                progressTracker.loadingDataPercent.set(rowsProcessed == totalRows ? 100 : (int) (rowsProcessed * 100.0 / totalRows));
```

Comment:
```suggestion
                progressTracker.loadingDataPercent.set(rowsProcessed >= totalRows ? 100 : (int) (rowsProcessed * 100.0 / totalRows));
```
To capture weird scenarios? It should NEVER happen, of course. Unsure if it happening is an indication of something horribly wrong.

Reply:
Indeed, this should never happen. I'm inclined to let progress go crazy if it means someone will report "progress is 2 billion percent, what is going on?" and we find out some horrible bug.

Reply:
Then again, if something like that blows we'll see it for sure in some different manner. No need to have an awkward progress. Fair enough, I'll take up the suggestion.